core/allocators.h

```diff
@@ -169,12 +160,10 @@ class BalloonAllocator {
 	}
 
 	void clear() {
-
+		Balloon *b;
```

Comment:
No reason to move this outside the loop. Reduces clarity

Reply:
Initially I moved it out as I thought It was an optimisation, but now I know that it isn't that simple. It does reduce clarity, and there probably isn't any optimisation considering it is a pointer type.
So I'll revert this back.